src/app/DTT/date/index.js

```diff
@@ -94,7 +94,7 @@ const Write = ({
   const [dateValue, setDateValue] = useState(null)
   const [isFocused, setIsFocused] = useState(false)
 
-  const onlyYear = type === 'year'
+  const onlyYear = dataType.component === 'year'
```

Comment:
dataType.component will always be date, component is not the key that differentiates between what format the data needs to be displayed. It's typeName. 
Hence from the logic you have put in, the variable onlyYear will always return false.
Check out the changes in this PR: https://github.com/genny-project/alyson/pull/872/files